back/engines/commercial/report_builder/app/services/report_builder/queries/registrations.rb

```diff
@@ -0,0 +1,67 @@
+module ReportBuilder
+  class Queries::Registrations < ReportBuilder::Queries::Base
+    def run_query(
+      start_at: nil,
+      end_at: nil,
+      resolution: 'month',
+      compare_start_at: nil,
+      compare_end_at: nil,
+      **_other_props
+    )
+      start_date, end_date = TimeBoundariesParser.new(start_at, end_at).parse
+      registrations_in_period = User.where(registration_completed_at: start_date..end_date)
+
```

Comment:
Since you insert `resolution` into an SQL query, it is probably best to validate it first (to avoid an SQL injection attack vector). e.g;
```
valid_resolutions = ['day', 'week', 'month', 'quarter', 'year']
  unless valid_resolutions.include?(resolution)
    raise ArgumentError, "Invalid resolution: #{resolution}. Must be one of: #{valid_resolutions.join(', ')}"
  end
  ```
  
  + some basic test(s) of this, obv.

Reply:
Ah yeah very good catch. Will add this 👍